Classes/Middleware/ElementBodyResponseMiddleware.php

```diff
@@ -84,6 +84,11 @@ private function extractElement(array $content, int $elementId, bool $recursiveE
         $body = [];
 
         foreach ($content as $items) {
+            // skip iteration if array is flat means doNotGroupByColPos = 1 is set
+            if(isset($items['id'])) {
```

Comment:
here you can just in use (int)($items['id'] ?? 0) === $elementId if true, array is flat & match id, then return $items, else search in nested colPos, just then condition should be after is_array check, then you can drop whole extra foreach after main one